metrics/firebase/test/firestore/rules/projects-test.js

```diff
@@ -29,37 +50,136 @@ describe("Projects collection rules", async function () {
+    await assertFails(
+      passwordProviderAllowedDomainApp
+        .collection(projectsCollectionName)
+        .doc("1")
+        .delete()
+    );
+  });
+
+  it("allows creating a project by an authenticated password user with not allowed email domain", async () => {
     await assertSucceeds(
-      authenticatedApp.collection(projectsCollectionName).add(project)
+      passwordProviderNotAllowedDomainApp
+        .collection(projectsCollectionName)
+        .add(project)
     );
   });
 
-  it("allows reading projects by an authenticated user", async () => {
+  it("allows reading projects by an authenticated password user with not allowed email domain", async () => {
     await assertSucceeds(
-      authenticatedApp.collection(projectsCollectionName).get()
+      passwordProviderNotAllowedDomainApp.collection(projectsCollectionName).get()
     );
   });
 
-  it("allows updating a project by an authenticated user", async () => {
+  it("allows updating a project by an authenticated password user with not allowed email domain", async () => {
     await assertSucceeds(
-      authenticatedApp
+      passwordProviderNotAllowedDomainApp
         .collection(projectsCollectionName)
         .doc("1")
         .update(project)
     );
   });
 
-  it("does not allow to delete a project by authenticated user", async () => {
+  it("does not allow to delete a project an authenticated password user with not allowed email domain", async () => {
     await assertFails(
-      authenticatedApp.collection(projectsCollectionName).doc("1").delete()
+      passwordProviderNotAllowedDomainApp
+        .collection(projectsCollectionName)
+        .doc("1")
+        .delete()
+    );
+  });
+
+  /**
+   * The google auth provider user specific tests
+   */
+
+  it("allows creating a project by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp.collection(projectsCollectionName).add(project)
+    );
+  });
+
+  it("allows reading projects by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp.collection(projectsCollectionName).get()
+    );
+  });
+
+  it("allows updating a project by an authenticated google user with allowed email domain", async () => {
+    await assertSucceeds(
+      googleProviderAllowedDomainApp
+        .collection(projectsCollectionName)
+        .doc("1")
+        .update(project)
+    );
+  });
+
+  it("does not allow to delete a project an authenticated google user with allowed email domain", async () => {
+    await assertFails(
+      googleProviderAllowedDomainApp
+        .collection(projectsCollectionName)
+        .doc("1")
+        .delete()
+    );
+  });
+
+  it("does not allow to create a project by an authenticated google user with not allowed email domain", async () => {
```

Comment:
```suggestion
  it("does not allow creating a project by an authenticated google user with not allowed email domain", async () => {
```
Here and in previous, and next tests.